homework/unique_ptr/tests/unique-test.cpp

```diff
@@ -0,0 +1,97 @@
+#include "gtest/gtest.h"
+#include "unique_ptr.hpp"
+
+class unique : public ::testing::Test {
+   protected:
+    int default_value{3};
+};
+
+TEST_F(unique, canBeDefaultCreated) {
+    cs::unique_ptr<int> ptr{};
+    EXPECT_EQ(ptr.get(), nullptr);
+}
+
+TEST_F(unique, canBeCreatedWithValue) {
+    cs::unique_ptr<int> ptr{new int{default_value}};
+    EXPECT_EQ(*ptr, default_value);
+}
+
+TEST_F(unique, canBeMovedByCtor) {
+    cs::unique_ptr<int> ptr{cs::unique_ptr<int>(new int{default_value})};
+    cs::unique_ptr<int> ptr2{std::move(ptr)};
+    EXPECT_EQ(*ptr2, default_value);
+    EXPECT_EQ(ptr.get(), nullptr);
+}
+
+TEST_F(unique, canBeMovedByMoveOperator) {
+    cs::unique_ptr<int> ptr{new int{default_value}};
+    cs::unique_ptr<int> ptr2 = std::move(ptr);
+    EXPECT_EQ(*ptr2, default_value);
+    EXPECT_EQ(ptr.get(), nullptr);
+}
+
+TEST_F(unique, canBeDereferenced) {
+    cs::unique_ptr<int> ptr{new int{default_value}};
+    EXPECT_EQ(*ptr, default_value);
+}
+
+TEST_F(unique, arrowOperatorCanBeUsed) {
+    struct foo {
+        void use(){};
+    };
+    cs::unique_ptr<foo> ptr{new foo()};
+    ptr->use();
```

Comment:
```suggestion
    EXPECT_NO_THROW(ptr->use());
```
You can use EXPECT_NO_THROW. Without this, in fact, this test do not test anything (or at least do not use any assert/expect) ;)

Reply:
Thanks!